js/storage.js

```diff
@@ -0,0 +1,33 @@
+const form = document.querySelector('.form-container');
+const { firstName, lastName, mail, msg } = form.elements;
+
+form.addEventListener('change', () => {
+  const obj = {};
+
+  if (firstName.value.trim() !== '') {
+    obj.firstName = firstName.value.trim();
+  }
+  if (lastName.value.trim() !== '') {
+    obj.lastName = lastName.value.trim();
+  }
+
+  if (mail.value.trim() !== '') {
+    obj.userEmail = mail.value.trim();
+  }
+
+  if (msg.value.trim() !== '') {
+    obj.userMessage = msg.value.trim();
+  }
+
+  localStorage.setItem('formData', JSON.stringify(obj));
+});
+
+// set the localStorage data to the form
+
+const formObj = JSON.parse(localStorage.getItem('formData'));
+const { userFirstname, userLastname, userMail, userrMsg } = formObj;
+
+firstName.value = userFirstname || '';
+lastName.value = userLastname || '';
+mail.value = userMail || '';
+msg.value = userrMsg || '';
```

Comment:
- [ ] Good job implementing the local storage, it is working perfectly 👏   But, you should consider double-checking how the data from the local storage is being called, so it gets displayed in the form's inputs after the user reloads the page, as required. Right now, when I reload the page, I can see it was stored, but nothing really shows up:
<img width="1440" alt="Screen Shot 2021-12-17 at 10 07 18 (2)" src="https://user-images.githubusercontent.com/66967274/146549553-b1774823-b97c-4bc2-a3c6-ce80f2fefbaa.png">

> You could consider something like `window.onload`. You can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload)